django_project/feti/management/commands/scraping_occupation.py

```diff
@@ -61,153 +158,193 @@ def create_occupation(data):
+                is_empty = True
+                print('No records found')
+                break
+            # get occupation information
+            if item.find("a", {"class": "linkNoMore"}):
+                pass
+            else:
+                print('Parsing the occupation list html...')
+                occupation = dict()
+                occupation['occupation'] = cleaning(item.a.string.split("Occupation Code")[0][:-1])
+                occupation['green_occupation'] = False
+                occupation['scarce_skill'] = False
+                # checking green
+                if item.find("span", {"class": "greenTagListView"}):
+                    occupation['green_occupation'] = True
+                if item.find("span", {"class": "scarceTagListView"}):
+                    occupation['scarce_skill'] = True
+
+                # get detail
+                print('Open learning pathway...')
+                html = get_soup(item.a.get('href'))
+                body = html.find("div", {"class": "BodyPanel642"})
+                if body:
+                    print('Parsing learning pathway...')
+                    content = str(body)
+                    content = content.split("<b>Description</b>")
+                    if len(content) > 1:
+                        content = content[1]
+                        # description
+                        occupation['description'] = cleaning(beautify(content.split("<b>")[0]).get_text())
+
+                        # tasks
+                        splits = content.split("<b>Tasks</b>")
+                        occupation['tasks'] = beautify(splits[1].split("<b>")[0]).get_text() \
+                            if len(splits) > 1 else ""
+                        occupation['tasks'] = cleaning(occupation['tasks'])
+
+                        # Occupation Regulation
+                        splits = content.split("<b>Occupation Regulation</b>")
+                        occupation['occupation_regulation'] = beautify(splits[1].split("<b>")[0]) \
+                            .get_text() \
+                            if len(splits) > 1 else ""
+                        occupation['occupation_regulation'] = cleaning(occupation['occupation_regulation'])
+
+                        # Learning Pathway Description
+                        splits = content.split("<b>Learning Pathway Description</b>")
+                        occupation['learning_pathway_description'] = beautify(
+                            splits[1].split("<a")[0]).get_text() \
+                            if len(splits) > 1 else ""
+                        occupation['learning_pathway_description'] = cleaning(
+                            occupation['learning_pathway_description'])
+
+                        # get learning pathway
+                        pathway_button = html.find("a", {"class": "btn_showLearningPathway"})
+                        occupation['learning_pathways'] = []
+                        if pathway_button:
```

Comment:
We are starting to nest too deeply here. Let's move this out to its own function.